File: csharp-sdk/src/Cleanster/Models/ChecklistItem.cs
using System.Text.Json.Serialization;

namespace Cleanster.Models;

/// <summary>A single task item within a <see cref="Checklist"/>.</summary>
public sealed record ChecklistItem
{
    [JsonPropertyName("id")]          public int     Id          { get; init; }
    [JsonPropertyName("description")] public string  Description { get; init; } = "";
    [JsonPropertyName("isCompleted")] public bool    IsCompleted { get; init; }
    /// <summary>URL of the proof photo uploaded by the cleaner, or <see langword="null"/> if not uploaded.</summary>
    [JsonPropertyName("imageUrl")]    public string? ImageUrl    { get; init; }
}

File: php-sdk/src/Models/ChecklistItem.php
<?php

declare(strict_types=1);

namespace Cleanster\Models;

/** A single task item within a Checklist. */
final class ChecklistItem
{
    public readonly int     $id;
    public readonly string  $description;
    public readonly bool    $isCompleted;
    public readonly ?string $imageUrl;
    /** The original raw array returned by the API. */
    public readonly array   $raw;

    public function __construct(array $data)
    {
        $this->id          = (int)($data['id'] ?? 0);
        $this->description = (string)($data['description'] ?? '');
        $this->isCompleted = (bool)($data['isCompleted'] ?? false);
        $this->imageUrl    = isset($data['imageUrl']) && $data['imageUrl'] !== null
                             ? (string)$data['imageUrl'] : null;
        $this->raw         = $data;
    }
}
